presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -0,0 +1,273 @@
+
+        boolean rightWithNull = false;
+        int rightRealSize = 0;
+        for (int i = 0; i < rightSize; i++) {
+            if (rightArray.isNull(i)) {
+                rightWithNull = true;
+            }
+            else {
+                arrayBigIntRight[rightRealSize++] = rightArray.getLong(i, 0);
+            }
+        }
+        sortIfUnsorted(arrayBigIntRight, rightRealSize);
+
+        int leftPosition = 0;
+        int rightPosition = 0;
+        while (leftPosition < leftRealSize && rightPosition < rightRealSize) {
+            if (arrayBigIntLeft[leftPosition] < arrayBigIntRight[rightPosition]) {
+                leftPosition++;
+            }
+            else if (arrayBigIntRight[rightPosition] < arrayBigIntLeft[leftPosition]) {
+                rightPosition++;
+            }
+            else {
+                return true;
+            }
+        }
+        return leftWithNull || rightWithNull ? null : false;
+    }
+
+    @SqlNullable
+    @TypeParameter("E")
+    @SqlType(StandardTypes.BOOLEAN)
+    public Boolean arraysOverlap(
+            @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"E", "E"}) MethodHandle lessThanFunction,
+            @TypeParameter("E") Type type,
+            @SqlType("array(E)") Block leftArray,
+            @SqlType("array(E)") Block rightArray)
+    {
+        int leftPositionCount = leftArray.getPositionCount();
+        int rightPositionCount = rightArray.getPositionCount();
+
+        if (leftPositionCount == 0 || rightPositionCount == 0) {
+            return false;
+        }
+
+        if (leftPositions.length < leftPositionCount) {
+            leftPositions = new int[leftPositionCount];
+        }
+
+        if (rightPositions.length < rightPositionCount) {
+            rightPositions = new int[rightPositionCount];
+        }
+
+        for (int i = 0; i < leftPositionCount; i++) {
+            leftPositions[i] = i;
+        }
+        for (int i = 0; i < rightPositionCount; i++) {
+            rightPositions[i] = i;
+        }
+        IntArrays.quickSort(leftPositions, 0, leftPositionCount, intBlockCompare(type, leftArray));
+        IntArrays.quickSort(rightPositions, 0, rightPositionCount, intBlockCompare(type, rightArray));
+
+        int leftCurrentPosition = 0;
+        int rightCurrentPosition = 0;
+        while (leftCurrentPosition < leftPositionCount && rightCurrentPosition < rightPositionCount) {
+            int compareValue = type.compareTo(leftArray, leftPositions[leftCurrentPosition], rightArray, rightPositions[rightCurrentPosition]);
+            if (compareValue > 0) {
+                rightCurrentPosition++;
+            }
+            else if (compareValue < 0) {
+                leftCurrentPosition++;
+            }
+            else {
+                return true;
+            }
+        }
+        boolean leftWithNull = leftArray.isNull(leftPositions[leftPositionCount - 1]);
```

Comment:
This is a comparator we usually have for similar functions, I don't think we need to comment this additionally only here.